### PR TITLE
Bring back clang-tidy

### PR DESCRIPTION
**Pull request by nsmithtt**

### CODE REVIEW

File: include/ttmlir/Dialect/TTNN/Analysis/ShardSolver.h [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 #include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
 #include "ttmlir/Dialect/TTNN/Analysis/Edge.h"
+
+#include <llvm/ADT/StringMap.h>
```
Comment (nobradovictt) [inline include/ttmlir/Dialect/TTNN/Analysis/ShardSolver.h:11]: Is this really needed?

File: include/ttmlir/Dialect/TTNN/Utils/OptimizerOverrides.h [hunk 2]
```diff
@@ -5,9 +5,12 @@
 #ifndef TTMLIR_DIALECT_TTNN_UTILS_OPTIMIZEROVERRIDES_H
 #define TTMLIR_DIALECT_TTNN_UTILS_OPTIMIZEROVERRIDES_H
 
+#include <llvm/Support/CommandLine.h>
+
+#include <cstdint>
```
Comment (nobradovictt) [inline include/ttmlir/Dialect/TTNN/Utils/OptimizerOverrides.h:10]: Is cstdint needed?

**Review by nobradovictt — APPROVED** [review]

File: lib/Dialect/TTNN/Transforms/Optimizer.cpp [hunk 3]
```diff
@@ -13,8 +13,128 @@
...
+namespace impl {
+
+std::unique_ptr<::mlir::Pass> createTTNNOptimizer();
+std::unique_ptr<::mlir::Pass> createTTNNOptimizer(TTNNOptimizerOptions options);
+
+// Go through the ops, set sharding specs for each op based on sharding
+// analysis, by updating layout attribute of each op.
+//
+template <typename DerivedT>
+class TTNNOptimizerBase : public ::mlir::OperationPass<::mlir::ModuleOp> {
```
Comment (nobradovictt) [inline lib/Dialect/TTNN/Transforms/Optimizer.cpp:26]: Thanks for fixing pass def!

File: lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp [hunk 4]
```diff
@@ -233,14 +234,20 @@ struct ConvolutionToConv2dPattern
...
-      ConvolutionDimension::FEATURE};
+  static inline const std::vector<int64_t> conv2dLayout = {
+      ConvolutionDimension::BATCH,
+      SPATIAL_DIM_HEIGHT,
+      SPATIAL_DIM_WIDTH,
+      ConvolutionDimension::FEATURE,
+  };
   // OIHW
-  const std::vector<int64_t> conv2dKernelLayout = {
+  static inline const std::vector<int64_t> conv2dKernelLayout = {
```
Comment (rpavlovicTT) [inline lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp:244]: why is `inline` required?

**Review by rpavlovicTT — APPROVED** [review]

File: lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp (hunk 4 above)
Comment (nsmithtt) [reply lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp:244]: - clang-tidy is mad about `const` data members, which are generally not a good pattern, they should be made static in 99% of cases.
- By making it `static` you then must define the member outside the class definition: https://en.cppreference.com/w/cpp/language/static
- I was too lazy and slightly preferred having the definition inline so I marked it `inline`

**Review by mtopalovicTT — APPROVED** [review]

**Review by pilkicTT — APPROVED** [review]

**Pull request merged**